azure_functions_worker/bindings/shared_memory_data_transfer/file_accessor_unix.py

```diff
@@ -0,0 +1,162 @@
+
+
+class FileAccessorUnix(FileAccessor):
+    """
+    For accessing memory maps.
+    This implements the FileAccessor interface for Unix platforms.
+    """
+    def open_mem_map(
+            self,
+            mem_map_name: str,
+            mem_map_size: int,
+            access: int = mmap.ACCESS_READ) -> Optional[mmap.mmap]:
+        """
+        Note: mem_map_size = 0 means open the entire mmap.
+        """
+        if mem_map_name is None or mem_map_name == '':
+            raise Exception(
+                f'Cannot open memory map. Invalid name {mem_map_name}')
+        if mem_map_size < 0:
+            raise Exception(
+                f'Cannot open memory map. Invalid size {mem_map_size}')
+        fd = self._open_mem_map_file(mem_map_name)
+        if fd is None:
+            return None
+        mem_map = mmap.mmap(fd.fileno(), mem_map_size, access=access)
+        return mem_map
+
+    def create_mem_map(self, mem_map_name: str, mem_map_size: int) \
+            -> Optional[mmap.mmap]:
+        if mem_map_name is None or mem_map_name == '':
+            raise Exception(
+                f'Cannot create memory map. Invalid name {mem_map_name}')
+        if mem_map_size <= 0:
+            raise Exception(
+                f'Cannot create memory map. Invalid size {mem_map_size}')
+        file = self._create_mem_map_file(mem_map_name, mem_map_size)
+        if file is None:
+            return None
+        mem_map = mmap.mmap(file.fileno(), mem_map_size, mmap.MAP_SHARED,
+                            mmap.PROT_WRITE)
+        if self._is_mem_map_initialized(mem_map):
+            raise Exception(f'Memory map {mem_map_name} already exists')
+        self._set_mem_map_initialized(mem_map)
+        return mem_map
+
+    def delete_mem_map(self, mem_map_name: str, mem_map: mmap.mmap) -> bool:
+        if mem_map_name is None or mem_map_name == '':
+            raise Exception(
+                f'Cannot delete memory map. Invalid name {mem_map_name}')
+        try:
+            fd = self._open_mem_map_file(mem_map_name)
+            os.remove(fd.name)
+        except Exception as e:
+            # In this case, we don't want to fail right away but log that
+            # deletion was unsuccessful.
+            # These logs can help identify if we may be leaking memory and not
+            # cleaning up the created memory maps.
+            logger.error(f'Cannot delete memory map {mem_map_name} - {e}',
+                         exc_info=True)
+            return False
+        mem_map.close()
+        return True
+
+    def _open_mem_map_file(self, mem_map_name: str) -> Optional[BufferedRandom]:
+        """
+        Get the file descriptor of an existing memory map.
+        Returns the BufferedRandom stream to the file.
+        """
+        # Iterate over all the possible directories where the memory map could
+        # be present and try to open it.
+        for mem_map_temp_dir in consts.UNIX_TEMP_DIRS:
```

Comment:
I'm concerned about two things here:
* Are we doing the same bad check over and over? Can we learn from the previous pass? Maybe instead of using the constant, check a variable which is sorted based on the previous success.
* I'm not sure using a exception is the best way; can we use something like os.path.exists(file_path)?